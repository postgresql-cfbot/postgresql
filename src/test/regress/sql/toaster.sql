

CREATE TOASTER tsttoaster HANDLER default_toaster_handler;
CREATE TOASTER IF NOT EXISTS tsttoaster HANDLER default_toaster_handler;
CREATE TOASTER IF NOT EXISTS tsttoaster1 HANDLER default_toaster_handler;

COMMENT ON TOASTER tsttoaster IS 'tsttoaster is a clone of default toaster';


CREATE TABLE tsttsrtbl_failed (
	t text TOASTER tsttoaster TOASTER tsttoaster;
);

CREATE TABLE tsttsrtbl (
	t1 text TOASTER tsttoaster,
	t2 text,
	t3 int
);

SELECT attnum, attname, atttypid, attstorage, tsrname
    FROM pg_attribute LEFT OUTER JOIN pg_toaster t ON t.oid = atttoaster
    WHERE attrelid = 'tsttsrtbl'::regclass and attnum>0
    ORDER BY attnum;

ALTER TABLE tsttsrtbl ALTER COLUMN t2 SET TOASTER tsttoaster;

SELECT attnum, attname, atttypid, attstorage, tsrname
    FROM pg_attribute LEFT OUTER JOIN pg_toaster t ON t.oid = atttoaster
    WHERE attrelid = 'tsttsrtbl'::regclass and attnum>0
    ORDER BY attnum;

ALTER TABLE tsttsrtbl ALTER COLUMN t3  TYPE text;

SELECT attnum, attname, atttypid, attstorage, tsrname
    FROM pg_attribute LEFT OUTER JOIN pg_toaster t ON t.oid = atttoaster
    WHERE attrelid = 'tsttsrtbl'::regclass and attnum>0
    ORDER BY attnum;

ALTER TABLE tsttsrtbl ALTER COLUMN t3  TYPE int USING t3::int;

\d+ tsttsrtbl

