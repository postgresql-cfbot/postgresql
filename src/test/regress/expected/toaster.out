CREATE TOASTER tsttoaster HANDLER default_toaster_handler;
CREATE TOASTER IF NOT EXISTS tsttoaster HANDLER default_toaster_handler;
NOTICE:  toaster "tsttoaster" already exists, skipping
CREATE TOASTER IF NOT EXISTS tsttoaster1 HANDLER default_toaster_handler;
COMMENT ON TOASTER tsttoaster IS 'tsttoaster is a clone of default toaster';
CREATE TABLE tsttsrtbl_failed (
	t text TOASTER tsttoaster TOASTER tsttoaster;
);
ERROR:  multiple TOASTER clauses not allowed
CREATE TABLE tsttsrtbl (
	t1 text TOASTER tsttoaster,
	t2 text,
	t3 int
);
SELECT attnum, attname, atttypid, attstorage, tsrname
    FROM pg_attribute LEFT OUTER JOIN pg_toaster t ON t.oid = atttoaster
    WHERE attrelid = 'tsttsrtbl'::regclass and attnum>0
    ORDER BY attnum;
 attnum | attname | atttypid | attstorage |  tsrname   
--------+---------+----------+------------+------------
      1 | t1      |       25 | x          | tsttoaster
      2 | t2      |       25 | x          | deftoaster
      3 | t3      |       23 | p          | 
(3 rows)

ALTER TABLE tsttsrtbl ALTER COLUMN t2 SET TOASTER tsttoaster;
SELECT attnum, attname, atttypid, attstorage, tsrname
    FROM pg_attribute LEFT OUTER JOIN pg_toaster t ON t.oid = atttoaster
    WHERE attrelid = 'tsttsrtbl'::regclass and attnum>0
    ORDER BY attnum;
 attnum | attname | atttypid | attstorage |  tsrname   
--------+---------+----------+------------+------------
      1 | t1      |       25 | x          | tsttoaster
      2 | t2      |       25 | x          | tsttoaster
      3 | t3      |       23 | p          | 
(3 rows)

ALTER TABLE tsttsrtbl ALTER COLUMN t3  TYPE text;
SELECT attnum, attname, atttypid, attstorage, tsrname
    FROM pg_attribute LEFT OUTER JOIN pg_toaster t ON t.oid = atttoaster
    WHERE attrelid = 'tsttsrtbl'::regclass and attnum>0
    ORDER BY attnum;
 attnum | attname | atttypid | attstorage |  tsrname   
--------+---------+----------+------------+------------
      1 | t1      |       25 | x          | tsttoaster
      2 | t2      |       25 | x          | tsttoaster
      3 | t3      |       25 | x          | deftoaster
(3 rows)

ALTER TABLE tsttsrtbl ALTER COLUMN t3  TYPE int USING t3::int;
\d+ tsttsrtbl
                                        Table "public.tsttsrtbl"
 Column |  Type   | Collation | Nullable | Default | Storage  |  Toaster   | Stats target | Description 
--------+---------+-----------+----------+---------+----------+------------+--------------+-------------
 t1     | text    |           |          |         | extended | tsttoaster |              | 
 t2     | text    |           |          |         | extended | tsttoaster |              | 
 t3     | integer |           |          |         | plain    |            |              | 

